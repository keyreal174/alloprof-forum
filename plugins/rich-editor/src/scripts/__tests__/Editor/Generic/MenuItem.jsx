import React from "react";
import { shallow } from "enzyme";
import MenuItem from "../../../Editor/Generic/MenuItem";

test("matches snapshot", () => {
    const clickHandler = () => {};
    const item = shallow(
        <MenuItem clickHandler={clickHandler} propertyName="bold" label="Bold" isActive={true} isFirst={false} isLast={false}/>
    );

    expect(item).toMatchSnapshot();
});

test("click handler is called", () => {
    const mockClickHandler = jest.fn();
    const item = shallow(
        <MenuItem clickHandler={mockClickHandler} propertyName="bold" label="Bold" isActive={true} isFirst={false} isLast={false}/>
    );
    item.find(".richEditor-button").simulate("click");
    expect(mockClickHandler.mock.calls.length).toBe(1);
});
