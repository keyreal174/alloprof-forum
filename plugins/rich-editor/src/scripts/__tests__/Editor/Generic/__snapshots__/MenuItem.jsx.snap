// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuItem
    clickHandler={[Function]}
    isActive={true}
    isFirst={false}
    isLast={false}
    label="Bold"
    propertyName="bold"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "richEditor.menu.bold",
      "aria-pressed": true,
      "children": <bold />,
      "className": "richEditor-button richEditor-menuItem isActive",
      "onBlur": [Function],
      "onClick": [Function],
      "onKeyDown": [Function],
      "role": undefined,
      "type": "button",
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "richEditor.menu.bold",
        "aria-pressed": true,
        "children": <bold />,
        "className": "richEditor-button richEditor-menuItem isActive",
        "onBlur": [Function],
        "onClick": [Function],
        "onKeyDown": [Function],
        "role": undefined,
        "type": "button",
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
